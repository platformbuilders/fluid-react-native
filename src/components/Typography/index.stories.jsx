import React from 'react';
import { storiesOf } from '@storybook/react-native';
import { withKnobs, text, select } from '@storybook/addon-knobs';
import Text from './index';
import typographyMarkdown from './typography.md';

const optionsVariants = [
  'largeTitle',
  'title1',
  'title2',
  'title3',
  'title4',
  'headline',
  'body',
  'subhead',
  'footnote',
  'caption1',
  'caption2',
];

storiesOf('Typography', module)
  .addDecorator(withKnobs)
  .addParameters({
    readme: {
      content: typographyMarkdown,
    },
    viewport: {
      defaultViewport: '',
    },
  })
  .add('Example', () => (
    <Text variant={select('variant', optionsVariants, 'title1')}>
      {text('children', 'Title 1')}
    </Text>
  ));
