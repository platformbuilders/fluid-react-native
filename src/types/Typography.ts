import { TypographyVariants } from './Variants';

type TypographyStyle = {
  fontSize: number;
  lineHeight: number;
};

export type TypographyTheme = {
  largeTitle: TypographyStyle;
  title1: TypographyStyle;
  title2: TypographyStyle;
  title3: TypographyStyle;
  headline: TypographyStyle;
  body: TypographyStyle;
  callout: TypographyStyle;
  subhead: TypographyStyle;
  footnote: TypographyStyle;
  caption1: TypographyStyle;
  caption2: TypographyStyle;
};

export interface TypographyType {
  variant?: TypographyVariants;
  children?: string | string[] | any;
  textRef?: any;
  id?: string;
  accessibility?: string;
  numberOfLines?: number;
}
