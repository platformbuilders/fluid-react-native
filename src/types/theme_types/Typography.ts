import { TypographyVariants } from './Variants';

export type TypographyTheme = {
  largeTitle: number;
  title1: number;
  title2: number;
  title3: number;
  headline: number;
  body: number;
  callout: number;
  subhead: number;
  footnote: number;
  caption1: number;
  caption2: number;
};

export interface TypographyType {
  variant?: TypographyVariants;
  children?: string | string[] | any;
  textRef?: any;
  id?: string;
  accessibility?: string;
  numberOfLines?: number;
}
