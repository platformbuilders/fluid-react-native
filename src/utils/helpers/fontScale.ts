import { RFValue } from 'react-native-responsive-fontsize';

export const scaledFontSize = (
  size: number | string,
  baselineFactor?: 280,
): number => {
  const value = typeof size === 'string' ? parseInt(size, 10) : size;
  return RFValue(value, baselineFactor);
};
