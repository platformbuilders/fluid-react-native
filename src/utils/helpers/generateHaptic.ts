import Haptic, { HapticFeedbackType } from 'react-native-haptic';
import { isIOS } from './isIOS';

export const generateHaptic = (haptic: HapticFeedbackType = 'impact'): void => {
  if (isIOS()) {
    Haptic.generate(haptic);
  }
};
