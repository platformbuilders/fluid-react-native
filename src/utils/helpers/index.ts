export {
  currencyParser,
  parseToThousands,
  removeWhiteSpaces,
  toOnlyNumbers,
  getTheme,
  ifStyle,
  switchStyle,
} from '@platformbuilders/helpers';
export * from './generateHaptic';
export * from './getByTypographyVariant';
export * from './fontScale';
export * from './shadow';
