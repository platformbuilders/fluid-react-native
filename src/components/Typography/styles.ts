import styled from 'styled-components/native';
import { getTheme, getFontSize, getLineHeight } from '../../utils/helpers';
import { TypographyType } from '../../types';

export const Text = styled.Text<TypographyType>`
  color: ${getTheme('text')};
  font-size: ${getFontSize}px;
  line-height: ${getLineHeight}px;
`;
