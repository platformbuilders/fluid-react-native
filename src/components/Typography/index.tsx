import React, { FC } from 'react';
import { TypographyType } from '../../types';
import { Text } from './styles';

const Typography: FC<TypographyType> = ({
  textRef = React.createRef(),
  variant = 'body',
  children,
  id,
  accessibility,
  ...rest
}) => (
  <Text
    testID={id || accessibility || `${children}`}
    accessibility={accessibility || `${children}`}
    accessibilityLabel={accessibility || `${children}`}
    ref={textRef}
    variant={variant}
    {...rest}
  >
    {children}
  </Text>
);

export default Typography;
