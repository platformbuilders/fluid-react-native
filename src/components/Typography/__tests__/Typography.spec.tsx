import 'jest';
import React from 'react';
import renderer from 'react-test-renderer';
import { ThemeProvider } from 'styled-components/native';
import Typography from '..';
import { theme } from '../../../test/helpers';

describe('<Typography />', () => {
  it('should render and match snapshot for default props', () => {
    const render = renderer.create(
      <ThemeProvider theme={theme}>
        <Typography accessibility="" />
      </ThemeProvider>,
    );
    expect(render.toJSON()).toMatchSnapshot();
  });

  it('should render and match snapshot for variant largeTitle', () => {
    const render = renderer.create(
      <ThemeProvider theme={theme}>
        <Typography accessibility="" variant="largeTitle" />
      </ThemeProvider>,
    );
    expect(render.toJSON()).toMatchSnapshot();
  });

  it('should render and match snapshot for variant title1', () => {
    const render = renderer.create(
      <ThemeProvider theme={theme}>
        <Typography accessibility="" variant="title1" />
      </ThemeProvider>,
    );
    expect(render.toJSON()).toMatchSnapshot();
  });

  it('should render and match snapshot for variant title2', () => {
    const render = renderer.create(
      <ThemeProvider theme={theme}>
        <Typography accessibility="" variant="title2" />
      </ThemeProvider>,
    );
    expect(render.toJSON()).toMatchSnapshot();
  });

  it('should render and match snapshot for variant title3', () => {
    const render = renderer.create(
      <ThemeProvider theme={theme}>
        <Typography accessibility="" variant="title3" />
      </ThemeProvider>,
    );
    expect(render.toJSON()).toMatchSnapshot();
  });

  it('should render and match snapshot for variant title4', () => {
    const render = renderer.create(
      <ThemeProvider theme={theme}>
        <Typography accessibility="" variant="title4" />
      </ThemeProvider>,
    );
    expect(render.toJSON()).toMatchSnapshot();
  });

  it('should render and match snapshot for variant headline', () => {
    const render = renderer.create(
      <ThemeProvider theme={theme}>
        <Typography accessibility="" variant="headline" />
      </ThemeProvider>,
    );
    expect(render.toJSON()).toMatchSnapshot();
  });

  it('should render and match snapshot for variant subhead', () => {
    const render = renderer.create(
      <ThemeProvider theme={theme}>
        <Typography accessibility="" variant="subhead" />
      </ThemeProvider>,
    );
    expect(render.toJSON()).toMatchSnapshot();
  });

  it('should render and match snapshot for variant footnote', () => {
    const render = renderer.create(
      <ThemeProvider theme={theme}>
        <Typography accessibility="" variant="footnote" />
      </ThemeProvider>,
    );
    expect(render.toJSON()).toMatchSnapshot();
  });

  it('should render and match snapshot for variant caption1', () => {
    const render = renderer.create(
      <ThemeProvider theme={theme}>
        <Typography accessibility="" variant="caption1" />
      </ThemeProvider>,
    );
    expect(render.toJSON()).toMatchSnapshot();
  });

  it('should render and match snapshot for variant caption2', () => {
    const render = renderer.create(
      <ThemeProvider theme={theme}>
        <Typography accessibility="" variant="caption2" />
      </ThemeProvider>,
    );
    expect(render.toJSON()).toMatchSnapshot();
  });
});
