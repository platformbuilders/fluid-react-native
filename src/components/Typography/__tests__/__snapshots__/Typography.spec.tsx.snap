// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Typography /> should render and match snapshot for default props 1`] = `
<Text
  style={
    Array [
      Object {
        "color": "#212121",
        "fontSize": 31,
        "lineHeight": 43,
      },
      Object {},
    ]
  }
  variant="body"
/>
`;

exports[`<Typography /> should render and match snapshot for variant caption1 1`] = `
<Text
  style={
    Array [
      Object {
        "color": "#212121",
        "fontSize": 24,
        "lineHeight": 31,
      },
      Object {},
    ]
  }
  variant="caption1"
/>
`;

exports[`<Typography /> should render and match snapshot for variant caption2 1`] = `
<Text
  style={
    Array [
      Object {
        "color": "#212121",
        "fontSize": 22,
        "lineHeight": 26,
      },
      Object {},
    ]
  }
  variant="caption2"
/>
`;

exports[`<Typography /> should render and match snapshot for variant footnote 1`] = `
<Text
  style={
    Array [
      Object {
        "color": "#212121",
        "fontSize": 26,
        "lineHeight": 35,
      },
      Object {},
    ]
  }
  variant="footnote"
/>
`;

exports[`<Typography /> should render and match snapshot for variant headline 1`] = `
<Text
  style={
    Array [
      Object {
        "color": "#212121",
        "fontSize": 33,
        "lineHeight": 43,
      },
      Object {},
    ]
  }
  variant="headline"
/>
`;

exports[`<Typography /> should render and match snapshot for variant largeTitle 1`] = `
<Text
  style={
    Array [
      Object {
        "color": "#212121",
        "fontSize": 67,
        "lineHeight": 80,
      },
      Object {},
    ]
  }
  variant="largeTitle"
/>
`;

exports[`<Typography /> should render and match snapshot for variant subhead 1`] = `
<Text
  style={
    Array [
      Object {
        "color": "#212121",
        "fontSize": 29,
        "lineHeight": 39,
      },
      Object {},
    ]
  }
  variant="subhead"
/>
`;

exports[`<Typography /> should render and match snapshot for variant title1 1`] = `
<Text
  style={
    Array [
      Object {
        "color": "#212121",
        "fontSize": 55,
        "lineHeight": 67,
      },
      Object {},
    ]
  }
  variant="title1"
/>
`;

exports[`<Typography /> should render and match snapshot for variant title2 1`] = `
<Text
  style={
    Array [
      Object {
        "color": "#212121",
        "fontSize": 43,
        "lineHeight": 55,
      },
      Object {},
    ]
  }
  variant="title2"
/>
`;

exports[`<Typography /> should render and match snapshot for variant title3 1`] = `
<Text
  style={
    Array [
      Object {
        "color": "#212121",
        "fontSize": 39,
        "lineHeight": 49,
      },
      Object {},
    ]
  }
  variant="title3"
/>
`;

exports[`<Typography /> should render and match snapshot for variant title4 1`] = `
<Text
  style={
    Array [
      Object {
        "color": "#212121",
        "fontSize": 35,
        "lineHeight": 43,
      },
      Object {},
    ]
  }
  variant="title4"
/>
`;
